### PR TITLE
Helper Function for Config Space Conversion

### PR DESCRIPTION
**Pull request by wistuba**
Simple helper function to convert a config space to dict and back.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/config_space.py [hunk 1]
```diff
@@ -1281,6 +1281,25 @@ def from_dict(d: dict) -> Domain:
     return domain
 
 
+def config_space_to_dict(
```
Comment (geoalgo) [inline syne_tune/config_space.py:1284]: Technically, we are converting to a dict so the name may a bit ambiguous (I took me some time to realize what this function was doing.

 Do you think `config_space_to_json_dict` (or `to_json_dict`) would be a better name? (and also config_space_from_dict -> `config_space_from_json_dict` or `from_json_dict`)

File: tst/test_config_space.py [hunk 2]
```diff
@@ -132,6 +134,19 @@ def test_serialization():
...
 
+def test_config_space_from_and_to_dict():
+    config_space = {
+        "int": randint(1, 2),
+        "logint": lograndint(3, 4),
+        "float": uniform(5.5, 6.5),
+        "logfloat": loguniform(7.5, 8.5),
+        "categorical": choice(["a", "b", "c"]),
+        "normal": randn(2.0, 1.0),
+        "const_str": "constant",
```
Comment (geoalgo) [inline tst/test_config_space.py:145]: We would probably want to add an integer and a float to cover all the union types.

**Review by geoalgo — COMMENTED** [review]
Very nice pull-request, I have just two minor comments.

File: tst/test_config_space.py (hunk 2 above)
Comment (wistuba) [reply tst/test_config_space.py:145]: added both cases

File: syne_tune/config_space.py (hunk 1 above)
Comment (geoalgo) [reply syne_tune/config_space.py]: Makes sense, I agree with that `to_json_dict` would be a better name.

**Review by geoalgo — APPROVED** [review]

**Pull request merged**